homeassistant/components/sentry/__init__.py

```diff
@@ -204,7 +206,7 @@ def process_before_send(
                 "channel": channel,
                 "custom_components": "\n".join(sorted(custom_components)),
                 "integrations": "\n".join(sorted(integrations)),
-                **system_info,
+                **system_info,  # type: ignore[arg-type]
```

Comment:
This probably can be solved by defining `system_info` as `Mapping`.

Reply:
A bit harder than that. `Mapping` is not needed:
```python
    ha_context: dict[str, dict[str, str | bool]] = {
        "Home Assistant": {
            "channel": channel,
            "custom_components": "\n".join(sorted(custom_components)),
            "integrations": "\n".join(sorted(integrations)),
            **system_info,
        },
    }
    event.setdefault("contexts", {}).update(ha_context)
```

Reply:
It is a documented issue, forcing the type there isn't a correct solution either (it is working around an open issue in mypy).

Reply:
Well, I think workaround is better than ignore 🙂

Reply:
Both aren't actually resulting in a proper check